torch/_inductor/fx_passes/pre_grad.py

```diff
@@ -71,9 +82,33 @@ def pre_grad_passes(gm, example_inputs):
     gm.graph.lint()
     gm.recompile()
 
+    # Check if there is still unbacked SymInts after dyn_rewritten_pass.
+    # Trigger RestartAnalysis to graph break since Inductor can't handle them.
+    if has_unbacked_symint(gm, example_inputs):
+        raise UnbackedSymIntError()
+
     return gm
 
 
+def has_unbacked_symint(gm, example_inputs):
+    fake_mode = detect_fake_mode(example_inputs)
+    for node in gm.graph.nodes:
+        if (
+            isinstance(node, torch.fx.node.Node)
+            and "example_value" in node.meta
+            and isinstance(node.meta["example_value"], FakeTensor)
+        ):
+            for x in torch.fx.experimental.symbolic_shapes.free_symbols(
+                node.meta["example_value"]
+            ):
+                if fake_mode.shape_env.is_unbacked_symint_or_symbool(x):
+                    for i in fake_mode.shape_env.runtime_var_to_range.keys():
+                        if str(i) == str(x):
```

Comment:
?? This seems like a very strange way to do this. Why do you have to do this test on top of the unbacked symint/symbool check?

Reply:
Yes, it's a typo, will update it.